app/code/Magento/Wishlist/Test/Unit/Controller/Index/UpdateTest.php

```diff
@@ -0,0 +1,252 @@
+                    'setQty',
+                    'save',
+                    'getDescription',
+                    'setDescription',
+                    'getProduct',
+                    'getName'
+                ]
+            )->getMock();
+
+        $this->objectManagerMock->expects($this->once())
+            ->method('create')
+            ->with(Item::class)
+            ->willReturn($itemMock);
+        $itemMock->expects($this->once())
+            ->method('load')
+            ->with(1)
+            ->willReturnSelf();
+        $itemMock->expects($this->once())
+            ->method('getWishLIstId')
+            ->willReturn($wishlistDataProvider['wishlist_data']['id']);
+        $itemMock->expects($this->once())
+            ->method('getDescription')
+            ->willReturn('');
+        $itemMock->expects($this->once())
+            ->method('setDescription')
+            ->willReturnSelf();
+        $itemMock->expects($this->once())
+            ->method('setQty')
+            ->willReturnSelf();
+        $dataMock = $this->createMock(Data::class);
+
+        $this->objectManagerMock->expects($this->exactly(2))
+            ->method('get')
+            ->with(Data::class)
+            ->willReturn($dataMock);
+        $dataMock->expects($this->once())
+            ->method('defaultCommentString')
+            ->willReturn('');
+        $dataMock->expects($this->once())
+            ->method('calculate');
+        $this->quantityProcessor->expects($this->once())
+            ->method('process')
+            ->willReturn($wishlistDataProvider['post_data']['qty']);
+
+        $productMock = $this->getMockBuilder(\Magento\Catalog\Model\Product::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $itemMock->expects($this->once())
+            ->method('getProduct')
+            ->willReturn($productMock);
+        $productMock->expects($this->once())
+            ->method('getName')
+            ->willReturn('product');
+        $this->messageManager->expects($this->once())
+            ->method('addSuccessMessage');
+        $this->assertEquals($this->resultRedirect, $this->updateController->execute());
+    }
+
+    /**
+     * Check if wishlist not availbale, and exception is shown
+     */
+    public function testUpdateWithNotFoundException()
+    {
+        $this->formKeyValidator->expects($this->once())
+            ->method('validate')
+            ->willReturn(true);
+        $this->wishlistProvider->expects($this->once())
+            ->method('getWishlist')
+            ->willReturn(null);
+        $this->expectException(NotFoundException::class);
+        $this->updateController->execute();
+    }
+
+    /**
+     * Dataprovider for Update test
+     *
+     * @return array
+     */
+    public function getWishlistDataProvider(): array
+    {
+        return [
+            [
+                [
+                    'wishlist_data' => [
+                        'id' => 1,
```

Comment:
Replace magic numbers with const - for example `STUB_PRODUCT_ID`

Reply:
ok